### DISCUSSION TITLE
Getting TypeScript errors when invoking a server action with the `action` prop

### DISCUSSION THREAD

**Discussion by adamchipperfield** [vercel/next.js#53010] [Help]
### Summary

I'm trying out the (experimental) [Server Actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions). When I use the form element's [`action` prop to invoke a server action](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#action), TypeScript doesn't like it (understandably, because I'm passing a function when it's expecting a string).

![image](https://github.com/vercel/next.js/assets/4995166/e76e8b18-7ede-4eb9-991f-2f6559098beb)

Does anyone know how I can extend the type of the form to allow functions?

**Comment by devjiwonchoi** [comment]
I guess you've already tried type for form as `FormData`?
